src/app/task-details/task-details.component.ts

```diff
@@ -0,0 +1,87 @@
+import { CommonModule } from '@angular/common';
+import { Component, OnInit } from '@angular/core';
+import { TaskService } from '../service/task.service';
+import { Task } from '../models/task.model';
+import { ActivatedRoute, Router } from '@angular/router';
+import { FormBuilder, FormGroup, ReactiveFormsModule, Validators } from '@angular/forms';
+import { Subject, takeUntil } from 'rxjs';
+import { response } from 'express';
+import { HttpClientModule } from '@angular/common/http';
+
+@Component({
+  selector: 'tasko-task-details',
+  standalone: true,
+  imports: [CommonModule, ReactiveFormsModule, HttpClientModule],
+  templateUrl: './task-details.component.html',
+  styleUrl: './task-details.component.scss'
+})
+export class TaskDetailsComponent implements OnInit {
+  task?: Task;
+  taskForm : FormGroup ;
+  isEditMode= false;
+  private destroy$: Subject<void>= new Subject<void>;
+
+  constructor(private route: ActivatedRoute, private taskService: TaskService, private router: Router, private fb:FormBuilder){
+    this.taskForm = this.fb.group({
+      title: ['', Validators.required],
+      description: ['', Validators.required],
+      type: ['', Validators.required],
+      status: ['pending', Validators.required]
+    });
+  }
+
+  ngOnInit(): void {
+    const taskId = this.route.snapshot.paramMap.get('id');
+    if (taskId){
+      this.getTask(taskId);
+    }
+    
+  }
+  ngOnDestroy(): void {
+    this.destroy$.next();
+    this.destroy$.complete();
+  }
+  getTask(id:string){
+    this.taskService.getATask(id).pipe(takeUntil(this.destroy$)).subscribe({
+      next: (response: Task)=>{
+        this.task = response;
+        if (this.task){
```

Comment:
it should fetch a task if the call is successful, otherwise an error should be thrown if the task does not exist